### PR TITLE
Newsletters: add #39 (2019-03-26)

### PR DESCRIPTION
**Pull request by harding**

### CODE REVIEW

File: _includes/references.md [hunk 1]
```diff
@@ -44,7 +44,6 @@ for details --> {% endcomment %}
 {% comment %}<!-- Later link definitions supersede earlier definitions.
 When more recent information about a BIP is available not in the regular
 place, put links here. -->{% endcomment %}
-[BIP151]: https://gist.github.com/jonasschnelli/c530ea8421b8d0e80c51486325587c52
```
Comment (harding) [inline _includes/references.md]: Note: this gist previously pointed to the updated version of BIP151.  However, now its latest HEAD points to the v2 encrypted protocol, so I've dropped it so that we link to the old version of BIP151 instead.  An alternative would be for me to find the previous updated-151 version in the gist repository history (let me know if you want that; I have no preference here).

**Comment by moneyball** [comment]
tACK

File: _posts/en/newsletters/2019-03-26-newsletter.md [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+## News
+
+- **Version 2 P2P transport proposal:** Jonas Schnelli sent a proposed
+  BIP to the Bitcoin-Dev mailing list that specifies an algorithm to be
+  used to encrypt traffic between peers.  It also specifies some other
+  minor changes to the creation of protocol messages, such as allowing
+  peers to use bandwidth-saving short identifiers and eliminating the
+  SHA256-based checksum on messages, as the [AEAD][]-based encryption
+  scheme protects data integrity.  The proposal is meant to replace
+  [BIP151][] and it contains links to an example implementation for
```
[the remark this answers is not in the document]
Comment (harding) [reply _posts/en/newsletters/2019-03-26-newsletter.md]: There's a for loop in references.md that creates links for all BIP numbers 1 to 400 (covering the entire currently-assigned range) so that I don't have to manually create a ref for each new BIP number Luke assigns.  Yeah, I'm that lazy.  :-)

File: _posts/en/newsletters/2019-03-26-newsletter.md [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+    (and is operated) correctly).  The exception is the creation of the
+    initial onchain transaction and the possible need for Bob to create
+    a refund transaction: if the trustless exchange doesn't happen, Bob
+    will receive no compensation for the onchain transaction fees
+    required for both of those transactions.  According to the [Loop
+    documentation][], their implementation has Alice send Bob a small
+    trusted payment in advance of the trustless exchange as an act of
+    good faith and an assurance that the operation won't end up costing
+    Bob money.
+
```
[the remark this answers is not in the document]
Comment (harding) [reply _posts/en/newsletters/2019-03-26-newsletter.md]: Yes.  I'm editing to add "via LN" so that it's clearer.  Thanks for catching that confusing omission!

File: _posts/en/newsletters/2019-03-26-newsletter.md [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+- **Help test Bitcoin Core 0.18.0 RC2:** The second Release Candidate
+  (RC) for the next major version of Bitcoin Core has been [released][0.18.0].
+  Testing is still needed by organizations and experienced users who
+  plan to run the new version of Bitcoin Core in production.  Use [this
+  issue][Bitcoin Core #15555] for reporting feedback.
+
+## News
+
+- **Version 2 P2P transport proposal:** Jonas Schnelli sent a proposed
```
[the remark this answers is not in the document]
Comment (harding) [reply _posts/en/newsletters/2019-03-26-newsletter.md]: D'oh, can't believe I forgot to put a link.  I'm going to link to the mailing list thread because I want to encourage people to comment there for now.  If we link to it in the future, we may use the gist (but hopefully he'll have a new BIP number by then).

File: _posts/en/newsletters/2019-03-26-newsletter.md [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+Optech tracks statistics about segwit use on our [dashboard][optech
+dashboard]; another site tracking related statistics is [P2SH.info][].
+We see an average of about 200 outputs per block are sent to native
+segwit addresses.  Those outputs are then spent in about 10% of all
+Bitcoin transactions.  That makes payments to native segwit addresses
+more popular than almost all altcoins.
+
+![Screenshot of Optech Dashboard segwit usage stats](/img/posts/2019-03-segwit-usage.png)
+
+However, wallets supporting segwit can also receive payment to
```
Comment (jnewbery) [inline _posts/en/newsletters/2019-03-26-newsletter.md]: I think this paragraph would be a little confusing for people not familiar with P2SH-wrapped segwit. Do you think the following would be clearer:

```
However, wallets supporting segwit can also receive payment to
backwards-compatible _P2SH-wrapped segwit_ addresses.  We can't directly track how many P2SH address are P2SH-wrapped segwit, but
the statistics sites linked earlier do tell us how often users spend
bitcoins they received to a P2SH-wrapped segwit address.  Currently, that
averages at about 37% of transactions.  If adoption remained steady at
that level, it would represent a minimum average of 1,400 outputs per
block.  Likely, it's even higher now.
```

I don't understand what you mean by '37% of transactions. Are you saying that 37% of spends are from P2SH-wrapped segwit UTXOs? Or that 37% of P2SH outputs that are spent are P2SH-wrapped segwit?

Comment (harding) [reply _posts/en/newsletters/2019-03-26-newsletter.md]: I told @jnewbery wrong.  It's that 37% of transactions contain at least one input that spends from a nested-segwit output.

I'll be editing the paragraph to try to improve clarity.

**Comment by moneyball** [comment]
tACK a09d1114774c04e0c618670c77c7e3c051a2fcb6

File: _posts/en/newsletters/2019-03-26-newsletter.md [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+questions and answers from Bitcoin StackExchange, and a list of notable
+code changes in popular Bitcoin infrastructure projects.
+
+## Action items
+
+- **Help test Bitcoin Core 0.18.0 RC2:** The second Release Candidate
+  (RC) for the next major version of Bitcoin Core has been [released][0.18.0].
+  Testing is still needed by organizations and experienced users who
+  plan to run the new version of Bitcoin Core in production.  Use [this
+  issue][Bitcoin Core #15555] for reporting feedback.
```
Comment (practicalswift) [inline _posts/en/newsletters/2019-03-26-newsletter.md]: Please see this comment https://github.com/bitcoin/bitcoin/issues/15555#issuecomment-473919220:

> I'd advise against reporting issues here. They should go in their own report. Otherwise it is hard to follow-up on reports when everything is in the same thread.

Comment (jnewbery) [reply _posts/en/newsletters/2019-03-26-newsletter.md]: Thanks @practicalswift! I think it still makes sense to point testers to a single location to 'leave feedback'. If they need to open an new issue for an unreported problem, that issue instructs them to do so.

**Comment by harding** [comment]
Untested ACK 85d0e235db35f4c1d0a4244f4e06e0736053f17b Thanks!
